kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/Utilities.java

```diff
@@ -0,0 +1,247 @@
+        HADOOP_CONF_FILES.forEach(
+            confFile -> {
+              Path path = Paths.get(hadoopConfDir, confFile);
+              if (Files.exists(path)) {
+                try {
+                  addResourceMethod.invoke(path.toUri().toURL());
+                } catch (IOException e) {
+                  LOG.warn("Error adding Hadoop resource {}, resource was not added", path, e);
+                }
+              }
+            });
+      }
+
+      // set any Hadoop properties specified in the sink config
+      config.hadoopProps().forEach(setMethod::invoke);
+
+      LOG.info("Hadoop config initialized: {}", configClass.getName());
+      return result;
+    } catch (Exception e) {
+      LOG.warn(
+          "Hadoop found on classpath but could not create config, proceeding without config", e);
+    }
+    return null;
+  }
+
+  public static Object extractFromRecordValue(Object recordValue, String fieldName) {
+    List<String> fields = Splitter.on('.').splitToList(fieldName);
+    if (recordValue instanceof Struct) {
+      return valueFromStruct((Struct) recordValue, fields);
+    } else if (recordValue instanceof Map) {
+      return valueFromMap((Map<?, ?>) recordValue, fields);
+    } else {
+      throw new UnsupportedOperationException(
+          "Cannot extract value from type: " + recordValue.getClass().getName());
+    }
+  }
+
+  private static Object valueFromStruct(Struct parent, List<String> fields) {
+    Struct struct = parent;
+    for (int idx = 0; idx < fields.size() - 1; idx++) {
+      Object value = fieldValueFromStruct(struct, fields.get(idx));
+      if (value == null) {
+        return null;
+      }
+      Preconditions.checkState(value instanceof Struct, "Expected a struct type");
+      struct = (Struct) value;
+    }
+    return fieldValueFromStruct(struct, fields.get(fields.size() - 1));
+  }
+
+  private static Object fieldValueFromStruct(Struct struct, String fieldName) {
+    Field structField = struct.schema().field(fieldName);
+    if (structField == null) {
+      return null;
+    }
+    return struct.get(structField);
+  }
+
+  private static Object valueFromMap(Map<?, ?> parent, List<String> fields) {
+    Map<?, ?> map = parent;
+    for (int idx = 0; idx < fields.size() - 1; idx++) {
+      Object value = map.get(fields.get(idx));
+      if (value == null) {
+        return null;
+      }
+      Preconditions.checkState(value instanceof Map, "Expected a map type");
+      map = (Map<?, ?>) value;
+    }
+    return map.get(fields.get(fields.size() - 1));
+  }
+
+  public static TaskWriter<Record> createTableWriter(
+      Table table, String tableName, IcebergSinkConfig config) {
+    Map<String, String> tableProps = Maps.newHashMap(table.properties());
+    tableProps.putAll(config.writeProps());
+
+    String formatStr = tableProps.getOrDefault(DEFAULT_FILE_FORMAT, DEFAULT_FILE_FORMAT_DEFAULT);
+    FileFormat format = FileFormat.valueOf(formatStr.toUpperCase());
```

Comment:
There's also `FileFormat.fromString` that handles `toUppserCase`.

Reply:
Nice, thanks, I updated this.